File: odin/bound_check.c
#include <stdio.h>


int main() {
    int arr[100000] = {0};
    int num = 0;
    for(int i = 0; i < 100000; i++) {
        arr[i] = num;
        num++;
    }

    for(int i = 0; i < 100000; i++) {
        printf("%d\n", arr[i]);
    }

    return 0;
}

File: odin/bound_check.zig
const std = @import("std");

pub fn main() void {
    var arr: [100000]i32 = .{0} ** 100000;
    var num: i32 = 0;

    for (0..100000) |i| {
        arr[i] = num;
        num += 1;
    }
    for (arr) |c| {
        std.debug.print("{d}\n", .{c});
    }
}
